index.js

```diff
@@ -50,7 +51,19 @@ export const rovingIndex = ({element:rover, target:selector}) => {
         focusPreviousItem(rover)
         break
     }
-  })
+  }
+  rover.addEventListener('keydown', onKeydown)
+
+  const cleanup = _ => {
+    rover.removeEventListener('focusin', onFocusin)
+    rover.removeEventListener('keydown', onKeydown)
+    rover.removeEventListener('DOMNodeRemoved', cleanup)
+
+    state.delete(rover)
+    targets.forEach(a => a.tabIndex = '')
+  }
+
+  rover.addEventListener('DOMNodeRemoved', cleanup)
```

Comment:
```suggestion
  rover.addEventListener('DOMNodeRemovedFromDocument', cleanup)
```

during testing i noticed that if the parent element to a rover was removed, the removed events werent fired and cleanup was never called. went looking and found `DOMNodeRemovedFromDocument` which does fire even if it's a parent element that does the removing. seems more likely that a developer would remove an entire section from the dom instead of the specific roving element, so this change will help cover more removal use cases.

TIL about `DOMNodeRemovedFromDocument`!